sopel/modules/seen.py

```diff
@@ -63,9 +70,12 @@ def seen(bot, trigger):
 @plugin.require_chanmsg
 def note(bot, trigger):
     nick = trigger.nick
-    # as of Sopel 8, `trigger.time` is Aware, meaning we should store its value
-    # for timezone safety when comparing it later
-    bot.db.set_nick_value(nick, 'seen_timestamp', trigger.time.timestamp())
-    bot.db.set_nick_value(nick, 'seen_channel', trigger.sender)
-    bot.db.set_nick_value(nick, 'seen_message', trigger)
-    bot.db.set_nick_value(nick, 'seen_action', trigger.ctcp is not None)
+    try:
+        # as of Sopel 8, `trigger.time` is Aware, meaning we should store its value
+        # for timezone safety when comparing it later
+        bot.db.set_nick_value(nick, 'seen_timestamp', trigger.time.timestamp())
+        bot.db.set_nick_value(nick, 'seen_channel', trigger.sender)
+        bot.db.set_nick_value(nick, 'seen_message', trigger)
+        bot.db.set_nick_value(nick, 'seen_action', trigger.ctcp is not None)
+    except SQLAlchemyError:
+        logger.exception("Database write failed")
```

Comment:
Is this not just raising another exception that will still spam logs.

Reply:
It doesn't raise an exception, but it does log it and could result in comparable spam to the logs. My approach to #2305 has been biased in the direction of cutting down IRC-visible spam, but maybe this fix doesn't actually close that issue because of this point.

Reply:
This will at least remove the automatic reply to IRC when there is an uncaught exception from a plugin callable, that's still better.

Of course, the ChannelLogger my disagree here but, you know what: I still hate the ChannelLogger. 🙃